core/src/main/java/org/elasticsearch/search/suggest/completion/context/GeoContextMapping.java

```diff
@@ -254,7 +254,7 @@ protected GeoQueryContext fromXContent(QueryParseContext context) throws IOExcep
      */
     @Override
     public List<InternalQueryContext> toInternalQueryContexts(List<GeoQueryContext> queryContexts) {
-        List<InternalQueryContext> internalQueryContextList = new ArrayList<>();
+        List<InternalQueryContext> internalQueryContextList = new ArrayList<>(queryContexts.size());
```

Comment:
Indeed, this one cannot be reliably be determined after all. A list of items is added each loop, depending on possible neighbors based on precision. I'll remove it.